packages/web-react/src/types/fileUploader.ts

```diff
@@ -0,0 +1,107 @@
+import { MutableRefObject, ReactNode } from 'react';
+import {
+  SpiritButtonElementProps,
+  SpiritDivElementProps,
+  SpiritInputElementProps,
+  SpiritLItemElementProps,
+  SpiritUListElementProps,
+  Validation,
+  ValidationTextType,
+} from './shared';
+
+export type FileUploaderAttachmentComponentType = (props: FileUploaderAttachmentBaseProps) => ReactNode;
+export type FileQueueMapType = Map<string, File>;
+export type FileUploaderErrorCallbackType = (error: string) => void;
+
+export interface FileUploaderTextProps {
+  helperText?: string;
+  labelText?: string;
+  linkText?: string;
+}
+
+export interface FileUploaderHandlingProps {
+  addToQueue: (key: string, file: File) => FileQueueMapType;
+  clearQueue: () => void;
+  fileQueue: FileQueueMapType;
+  onDismiss: (key: string) => FileQueueMapType;
+}
+
+export interface FileUploaderErrorMessagesProps {
+  errorMessages?: {
+    errorFileDuplicity: string;
+    errorMaxFileSize: string;
+    errorMaxUploadedFiles: string;
+  };
+}
+
+export interface AttachmentDismissButtonBaseProps extends SpiritButtonElementProps {}
+
+export interface FileUploaderInputBaseProps
+  extends Omit<SpiritInputElementProps, 'onError'>,
+    FileUploaderTextProps,
+    Validation {
+  dropZoneRef?: MutableRefObject<HTMLDivElement>;
+  id: string;
+  inputRef?: MutableRefObject<HTMLInputElement>;
+  isDisabled?: boolean;
+  isMultiple?: boolean;
+  label?: string;
+  maxFileSize?: number;
+  maxUploadedFiles?: number;
+  name: string;
+  onError?: FileUploaderErrorCallbackType;
+  validationText?: ValidationTextType;
+}
+
+export interface FileUploaderListBaseProps extends SpiritUListElementProps {
+  attachmentComponent: FileUploaderAttachmentComponentType;
+  id: string;
+  inputName: string;
+  label?: string;
+}
+
+export interface FileUploaderAttachmentBaseProps extends Omit<SpiritLItemElementProps, 'onError'> {
+  buttonLabel?: string;
+  file: File;
+  id: string;
+  label: string;
+  name: string;
+  onDismiss: (key: string) => FileQueueMapType;
+  onError?: FileUploaderErrorCallbackType;
+}
+
+export interface FileUploaderBaseProps extends SpiritDivElementProps, Partial<FileUploaderErrorMessagesProps> {
+  id: string;
+  isDisabled?: boolean;
+}
+
+export interface UncontrolledFileUploaderBaseProps extends Omit<FileUploaderBaseProps, 'onChange'> {
```

Comment:
This shares some of the props of the `FileUploaderInputBaseProps`. Can we have some intermediate interface that has those shared props? Like `isMultiple` or `maxFileSize`, etc.

Reply:
Done, created `FileUploaderIntermediateProps`